### PR TITLE
⭐️ define proto for cnspec report

### PR DESCRIPTION
**Pull request by chris-rock**

### CODE REVIEW

File: cli/reporter/cnquery_report.proto [hunk 1]
```diff
@@ -0,0 +1,34 @@
...
+  string type = 3;
+  string platform_name = 4;
+}
+
+message DataValues {
+  map <string, DataValue> values = 1;
+}
+
+message DataValue {
+  string mrn = 1;
```
Comment (chris-rock) [inline cli/reporter/cnquery_report.proto]: remove mrn

File: apps/cnquery/cmd/plugin.go [hunk 2]
```diff
@@ -197,7 +197,7 @@ func (c *cnqueryPlugin) RunQuery(conf *run.RunQueryConfig, runtime *providers.Ru
 		if conf.Format != "json" {
 			sh.PrintResults(code, results)
 		} else {
-			reporter.BundleResultsToJSON(code, results, out)
+			_ = reporter.CodeBundleToJSON(code, results, out)
```
Comment (preslavgerchev) [inline apps/cnquery/cmd/plugin.go:200]: why discard the result here?

File: cli/reporter/proto.go [hunk 3]
```diff
@@ -0,0 +1,126 @@
...
+				queryMrnIdx[query.CodeId] = query.Mrn
+			}
+		}
+	}
+
+	// fill in assets
+	for assetMrn, a := range data.Assets {
+		pAsset := &Asset{
+			Mrn:  a.Mrn,
+			Name: a.Name,
```
Comment (preslavgerchev) [inline cli/reporter/proto.go:46]: do we also want the platform name and type here?

File: cli/reporter/cnquery_report.proto [hunk 4]
```diff
@@ -0,0 +1,33 @@
...
+message Report {
+  map <string, Asset> assets = 1;
+  map <string, DataValues> data = 2;
+  map <string, string> errors = 3;
+}
+
+message Asset {
+  string mrn = 1;
+  string name = 2;
+  string type = 3;
```
Comment (preslavgerchev) [inline cli/reporter/cnquery_report.proto]: whats type representing?

File: cli/reporter/proto.go [hunk 5]
```diff
@@ -0,0 +1,126 @@
...
+		assetMrn := prettyPrintString(id)
+		errorMsg := errStatus.Message
+		protoReport.Errors[assetMrn] = errorMsg
+	}
+
+	return protoReport, nil
+}
+
+func (r *Report) ToJSON() ([]byte, error) {
+	return protojson.Marshal(r)
```
Comment (preslavgerchev) [inline cli/reporter/proto.go:113]: does this intentionally use `protojson` instead of just `json?`

File: cli/reporter/proto.go (hunk 3 above)
Comment (chris-rock) [reply cli/reporter/proto.go:46]: We do not have that information right now, in cnquery asset. We can add that later if required.

File: cli/reporter/proto.go (hunk 5 above)
Comment (chris-rock) [reply cli/reporter/proto.go:113]: Should make it more safe for proto. Output should be the same.

File: apps/cnquery/cmd/plugin.go (hunk 2 above)
Comment (chris-rock) [reply apps/cnquery/cmd/plugin.go:200]: we only discard the error. output is rendered into out writer

File: cli/reporter/cnquery_report.proto (hunk 4 above)
Comment (chris-rock) [reply cli/reporter/cnquery_report.proto]: We do not use it right now, will remove

**Review by preslavgerchev — APPROVED** [review]

**Pull request merged**